rpcserver.go

```diff
@@ -7825,9 +7848,40 @@ func (r *rpcServer) rfqChannel(ctx context.Context, id asset.ID,
 
 	// If the user specified a peer public key, and we still have multiple
 	// channels, it means we have multiple channels with the same asset and
-	// the same peer. So we just return the first channel.
+	// the same peer.
 	case len(assetBalances) >= 1:
-		return &assetBalances[0], nil
+		// Initialize to first element.
+		bestBalance := assetBalances[0]
+
+		switch intention {
+		case ReceiveIntention:
```

Comment:
instead of nesting another `switch` here, I think you should return early above if `len(assetBalances) == 0`